modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/PrepareServiceImpl.java

```diff
@@ -296,4 +364,67 @@ private ResultSetMetadata resultSetMetadata(
                 }
         );
     }
+
+    private static SqlNode parse(String query, QueryContext queryContext, BaseQueryContext ctx) {
+        StatementParseResult parseResult = IgniteSqlParser.parse(query, StatementParseResult.MODE);
+
+        SqlNode sqlNode = parseResult.statement();
+
+        validateParsedStatement(queryContext, parseResult, sqlNode, ctx.parameters());
+
+        return sqlNode;
+    }
+
+    private static boolean skipCache(SqlNode sqlNode) {
+        SqlKind kind = sqlNode.getKind();
+
+        switch (kind) {
+            case SELECT:
+            case INSERT:
+                return false;
```

Comment:
I don't quite understand why we don't caching only 'INSERT' instead of caching all DML queries?

For example, I have the following results of `InsertBenchmark` if we changing `insert` query to something with UPDATE: `update usertable set field3='a' where ycsb_key=?`

Patch not applied:
```
Benchmark                  (fsync)  Mode  Cnt  Score   Error  Units
InsertBenchmark.sqlInsert    false  avgt       1.288          ms/op
```

Patch applied:
```
Benchmark                  (fsync)  Mode  Cnt  Score   Error  Units
InsertBenchmark.sqlInsert    false  avgt       3.553          ms/op
```

Reply:
There is an issue with DELETE statement https://issues.apache.org/jira/browse/IGNITE-19866.
Allow caching for the rest of DML queries.